src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/ArmorTask.java

```diff
@@ -89,25 +93,28 @@ private void handleSlimefunArmor(Player p, ItemStack[] armor, HashedArmorpiece[]
     }
 
     private void checkForSolarHelmet(Player p) {
-        // Temporary performance improvement
-        if (!SlimefunUtils.isItemSimilar(p.getInventory().getHelmet(), SlimefunItems.SOLAR_HELMET, true)) {
+        ItemStack helmet = p.getInventory().getHelmet();
+
+        if (SlimefunPlugin.getRegistry().isBackwardsCompatible() && !SlimefunUtils.isItemSimilar(helmet, SlimefunItems.SOLAR_HELMET, true, false)) {
+            // Performance saver for slow backwards-compatible versions of Slimefun
             return;
         }
 
-        SlimefunItem item = SlimefunItem.getByItem(p.getInventory().getHelmet());
+        SlimefunItem item = SlimefunItem.getByItem(helmet);
 
-        if (item instanceof SolarHelmet && Slimefun.hasUnlocked(p, item, true) && hasSunlight(p)) {
-            ItemEnergy.chargeInventory(p, (float) ((SolarHelmet) item).getChargeAmount());
+        if (item instanceof SolarHelmet && Slimefun.hasUnlocked(p, item, true)) {
+            ((SolarHelmet) item).rechargeItems(p);
         }
     }
 
     private boolean hasSunlight(Player p) {
-        return (p.getWorld().getTime() < 12300 || p.getWorld().getTime() > 23850) && p.getEyeLocation().getBlock().getLightFromSky() == 15;
+        World world = p.getWorld();
```

Comment:
I'd probably check if env is end (or check the default world) and return 0 (also, could return true in nether, not sunlight but it's supposed to be a "hell" thing. I like to do it this way but dealers choice).

This will be quicker than the block initialisation which is done below.

Reply:
??

Reply:
The end isn't ever light level 15 and considering initialising the block is slow anyway you may as well check the env (or the world) to just insta return false. Small optimisation but optimisations are good.

Reply:
I really have no idea what you mean, I know the Block call is slow but I don't understand what exact opimization you are trying to describe.
If you can outline it as code or make a follow up commit, it'd be awesome.

Reply:
if (world.getEnvironment() == Envrionment.END) return false;

Reply:
And true in the nether?
I am just confused as you said "return 0" when this is a boolean method.

Reply:
Shh.

And I'd say return true in nether since it's supposed to be a "hell" kinda thing. But, that's really up to you. Not sure the normal light levels of the nether but yeah. Just dealers choice in this regard.

Reply:
I'll probably make both false for now.